src/components/apps/editor/params/CheckboxPropertyFields.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * A form component for editing App `Flag` parameter properties.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import { FastField } from "formik";
+
+import { useTranslation } from "i18n";
+
+import ids from "../ids";
+
+import DescriptionField from "./common/DescriptionField";
+import LabelField from "./common/LabelField";
+import VisibleField from "./common/VisibleField";
+
+import {
+    build as buildID,
+    FormCheckbox,
+    FormTextField,
+} from "@cyverse-de/ui-lib";
+
+import { Grid } from "@material-ui/core";
+
+export default function CheckboxPropertyFields(props) {
+    const { baseId, fieldName } = props;
+
+    const { t } = useTranslation("app_editor");
+
+    return (
+        <Grid container>
+            <Grid item xs={12}>
+                <LabelField baseId={baseId} fieldName={fieldName} />
+            </Grid>
+            <Grid
+                item
+                xs={12}
+                container
+                direction="row"
+                justify="flex-start"
+                alignItems="flex-start"
+            >
+                <Grid item xs={6}>
+                    <FastField
+                        id={buildID(baseId, ids.PARAM_FIELDS.CHECKED_OPTION)}
+                        name={`${fieldName}.name.checked.option`}
+                        label={t("checkboxArgOptionCheckedLabel")}
+                        margin="normal"
+                        required
+                        component={FormTextField}
```

Comment:
I mentioned this in the PR description, but these Checkbox `option` fields are marked as `required`, though they don't have any form validation checks. So the user can leave both blank and still save the App (although it probably won't be able to run).

The old DE would display an error if both were left blank and not allow the user to save the App in this state. I'm not sure if we want to do the same thing here, or go ahead and allow the user to save a "broken" app (but we could at least provide some additional help text explaining that below these fields).